### PR TITLE
[healthprediction] Update maths [WIP]

### PR DESCRIPTION
**Pull request by ls-**
Fixes maths related to heal absorb and incoming heals values. Previously we're discarding the excess of heal absorbs, while it should've been used to adjust inc heals values.

~I'll also update element's docs tomorrow, an example of `PostUpdate` is required to show people how to properly imitate Blizz prediction bars' behaviour. In our case `absorbBar` needs to be dynamically appended to either health bar or `otherBar`, depending on which one is the rightmost.~

The main difference is that inc heals and heal absorbs are mutually exclusive now. If the amount of heal absorb is larger than the amount of inc heals, its value will be adjusted and inc heals will be discarded. and vice versa.

This issue was highlighted in #409.

### CODE REVIEW

**Comment by ls-** [comment]
Also, I may add one more arg for `PostUpdate` to show that inc heals are there, but they get absorbed, so devs could do something w/ it.

**Comment by ls-** [comment]
Meh, turns out we don't want to have mandatory `PostUpdate`s in oUF, so I'll rewrite it tomorrow.

**Comment by ls-** [comment]
The main difference is that inc heals and heal absorbs are mutually exclusive now. If the amount of heal absorb is larger than the amount of inc heals, its value will be adjusted and inc heals will be discarded. and vice versa.

This way PostUpdate-voodoo isn't needed.

File: elements/healthprediction.lua [hunk 1]
```diff
@@ -104,41 +104,40 @@ local function Update(self, event, unit)
...
+			otherIncomingHeal = allIncomingHeal - myIncomingHeal
 		end
 	end
 
-	if(healAbsorb > allIncomingHeal) then
-		healAbsorb = healAbsorb - allIncomingHeal
-	else
-		healAbsorb = 0
+	local hasOverAbsorb = false
+	if(health + allIncomingHeal + absorb >= maxHealth or health + absorb >= maxHealth) then
```
Comment (Rainrider) [inline elements/healthprediction.lua]: The `health + absorb >= maxHealth` condition is not needed. `allIncomingHeal` is always non-negative.

**Comment by ls-** [comment]
> Maybe add a note that incoming heals adjustments for heal absorb are at the cost of others' heals first and player's heals after that.

IMHO, not needed, it's kinda obvious from the code.

**Pull request merged**